### PR TITLE
Use route proxy as middleware

### LINKED ISSUE #50: Error on PUT and POST method
When  I send request from the browser using **the-fake-backend**  like a proxy, I receive this error:

`[HPM] Error occurred while trying to proxy request {endpoint} from localhost:8081 to {proxy} (ECONNRESET) (https://nodejs.org/api/errors.html#errors_common_system_errors)`

GET method works normally.

### PR DESCRIPTION
**Pull request by fsmaia**
Fixes #50.

This bug was due to `http-proxy-middleware` not being used as middleware.

This PR also updates `http-proxy-middleware` version.

### CODE REVIEW

File: source/server.ts [hunk 1]
```diff
@@ -54,6 +54,18 @@ export function createServer(options = {} as ServerOptions): Server {
     uiManager.drawRequest(req, res, next)
   );
 
+  expressServer.use((req: Request, res, next) => {
+    const route = routeManager.findRouteByPath(req.path.replace(basePath, ''));
+    if (route) {
+      const proxy = proxyManager.resolveRouteProxy(route);
+      if (proxy) {
+        return proxy?.(req, res, next);
```
Comment (rhberro) [inline source/server.ts:62]: You've already checked if proxy does exist, is `proxy?.` really necessary in here?

File: source/routes.ts [hunk 2]
```diff
@@ -130,4 +130,11 @@ export class RouteManager {
       ],
     });
   }
+
+  /**
+   * Find a route by path.
+   */
+  findRouteByPath(path: string) {
+    return this.routes.find((route) => route.path === path);
```
Comment (rhberro) [inline source/routes.ts:138]: We have `ramda` as a dependency, what do you think about using `this.routes.find(propEq('path', path));` instead?

File: source/ui.spec.ts [hunk 3]
```diff
@@ -55,7 +55,7 @@ describe('source/ui.ts', () => {
   describe('UIManager', () => {
     let uiManager: UIManager;
     const routeManager = new RouteManager();
-    const proxyManager = new ProxyManager([], routeManager);
+    const proxyManager = new ProxyManager(routeManager, []);
```
Comment (rhberro) [inline source/ui.spec.ts:58]: The second param has a default value, can we omit it here?

**Review by rhberro — APPROVED** [review]
Awesome! I've requested some minor changes that doesn't block your pull request of being merged. Thank you so much, I appreciate your help!

File: source/routes.ts (hunk 2 above)
Comment (rhberro) [reply source/routes.ts:138]: Not sure we should keep `ramda` as a dependency.

File: source/server.ts (hunk 1 above)
Comment (fsmaia) [reply source/server.ts:62]: Nope, but in the next upcoming pull request I fixed this.

File: source/routes.ts (hunk 2 above)
Comment (fsmaia) [reply source/routes.ts:138]: Nice idea.

File: source/ui.spec.ts (hunk 3 above)
Comment (fsmaia) [reply source/ui.spec.ts:58]: For sure.

**Review by rhberro — APPROVED** [review]

**Pull request merged**